### ISSUE TITLE
Borg Backup - Data integrity error: Archive authentication did not verify

### ISSUE THREAD

**Issue by ebildebil** [nextcloud/all-in-one#3336]
### Steps to reproduce
1. Run backup in AIO console or verify backup
2.
3.

### Expected behavior 
Backup to be successful as always.
### Actual behavior 

Receiving Archive Authentication errors. It's the same archive built and verified by AIO-Borg and has been running everyday for quite a long time.

I did update AIO master container and other containers today.

I have not clicked on repair archive yet since this type of error is a first. I have in the past used repair archive when correcting index errors or orphaned objects were found etc.

### Host OS 
Ubuntu Server 22.04, 
Kernel - 6.2.0-32-generic (to add 13-gen intel igpu drivers)
Docker on bare metal nuc 13, all SSD's.

Backups are performed on local ssd and then rsynced to NAS via Cron

#### Nextcloud AIO version 
7.1.1
#### Current channel 
Latest
#### Other valuable info 
Checking the backup integrity...
Starting repository check
finished segment check at segment 4485
Starting repository index check
Index object count match.
Finished full repository check, no problems found.
Starting archive consistency check...
Starting cryptographic data integrity verification...
Finished cryptographic data integrity verification, verified 501935 chunks with 0 integrity errors.
Analyzing archive 20230331_000122-nextcloud-aio (1/18)
Analyzing archive 20230430_000119-nextcloud-aio (2/18)
Analyzing archive 20230531_000138-nextcloud-aio (3/18)
Analyzing archive 20230630_000125-nextcloud-aio (4/18)
Analyzing archive 20230731_000135-nextcloud-aio (5/18)
Analyzing archive 20230813_000115-nextcloud-aio (6/18)
Archive TAM authentication issue for archive 20230813_000115-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230820_000129-nextcloud-aio (7/18)
Archive TAM authentication issue for archive 20230820_000129-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230827_000135-nextcloud-aio (8/18)
Analyzing archive 20230831_000146-nextcloud-aio (9/18)
Archive TAM authentication issue for archive 20230831_000146-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230903_000132-nextcloud-aio (10/18)
Analyzing archive 20230906_000139-nextcloud-aio (11/18)
Analyzing archive 20230907_000127-nextcloud-aio (12/18)
Archive TAM authentication issue for archive 20230907_000127-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230908_000128-nextcloud-aio (13/18)
Archive TAM authentication issue for archive 20230908_000128-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230909_000126-nextcloud-aio (14/18)
Archive TAM authentication issue for archive 20230909_000126-nextcloud-aio: Data integrity error: Archive authentication did not verify
Analyzing archive 20230909_212206-nextcloud-aio (15/18)
Analyzing archive 20230910_000138-nextcloud-aio (16/18)
Analyzing archive 20230911_000133-nextcloud-aio (17/18)
Analyzing archive 20230912_000156-nextcloud-aio (18/18)
4170 orphaned objects found!
Archive consistency check complete, problems found.
Some errors were found while checking the backup integrity!
Check the AIO interface for advices on how to proceed now!

**Comment by szaimen** [comment]
https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-2-5-archives-spoofing-vulnerability-cve-2023-36811

**Comment by szaimen** [comment]
Short question: did you get an error during normal daily backup or only by checking the integrity?

**Comment by ebildebil** [comment]
I received the error on both. 
The normal daily backup failed with the above error. After which I ran the integrity check and posted that result here.

I think I did read that link you posted earlier today, but couldn't conclude what to do with it.

**Comment by szaimen** [comment]
> I think I did read that link you posted earlier today, but couldn't conclude what to do with it.

Yeah I guess we will built the logic into our check-repair feature so that it gets automatically fixed if one should run into it.

**Comment by ebildebil** [comment]
Thank you! and Okay. So, in the meanwhile, standby by for further instructions ?

**Comment by szaimen** [comment]
Yes, one quesrion from my side, can you run the following command on the server and report back its output?

```
BORG_WORKAROUNDS=ignore_invalid_archive_tam borg info --debug <repo> 2>&1 | grep TAM | grep -i manifest
```

Of course <repo> needs to be the location where the archive is stored.

**Comment by ebildebil** [comment]
Sure. I'm on a bus now so it'll be sometime before I can.
Also is the command straight in terminal or via docker exec?

**Comment by szaimen** [comment]
Straight in terminal but of course borg needs to be installed first on the server if it is not yet

**Comment by ebildebil** [comment]
Okay, will do.

**Comment by ebildebil** [comment]
Hi, Just got back to my system.
Installed Borg via APT, (1.2.0)
below is the output:

`Enter passphrase for key /localbackups/nextcloud/borg:
TAM-verified manifest
Manifest is TAM verified and says TAM is required, updating security database...`

**Comment by szaimen** [comment]
Thanks! Can you run
```
BORG_WORKAROUNDS=ignore_invalid_archive_tam borg list --format='{name} {time} tam:{tam}{NL}' /localbackups/nextcloud/borg
```

**Comment by ebildebil** [comment]
I get the below output:

```
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5089, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5020, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 183, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 1336, in do_list
    return self._list_repository(args, repository, manifest, key)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 1379, in _list_repository
    sys.stdout.write(formatter.format_item(archive_info))
  File "/usr/lib/python3/dist-packages/borg/helpers/parseformat.py", line 550, in format_item
    return self.format.format_map(self.get_item_data(item))
KeyError: 'tam'

Platform: Linux bisu-mail-cloud 6.2.0-32-generic #32~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 18 10:40:13 UTC 2 x86_64
Linux: Unknown Linux
Borg: 1.2.0  Python: CPython 3.10.12 msgpack: 1.0.3 fuse: pyfuse3 3.2.0 [pyfuse3,llfuse]
PID: 4140333  CWD: /localbackups/nextcloud
sys.argv: ['/usr/bin/borg', 'list', '--format={name} {time} tam:{tam}{NL}', '/localbackups/nextcloud/borg']
SSH_ORIGINAL_COMMAND: None
```

**Comment by szaimen** [comment]
I see, so the command does not work with borg 1.2.0

**Comment by szaimen** [comment]
What I find confusing is that I cannot reproduce the issue on my test instance. 
Here are the possible explanations:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/b19694fc-960f-462e-9e52-f0e7fe336eb0)

AIO never runs borg rename or borg recreate. So the only valid explanation is this: 
>  “tam:none” could also come from archives created by an attacker. You should verify that “tam:none” archives are authentic and not malicious (== have good content, have correct timestamp, can be extracted successfully). In case you find crappy/malicious archives, you must delete them before proceeding.

So either the archives have been created by an attacker on your system or they are somehow crappy/malicious because they don't have good content, or correct timestamp or cannot be extracted successfully. 

My recommendation for you would be to delete the archives that are bad by hand by using borg delete `<archive>`.

Also see https://github.com/nextcloud/all-in-one#delete-backup-archives-manually
